python/ray/node.py

```diff
@@ -186,6 +186,12 @@ def __init__(self,
             redis_client.set("session_name", self.session_name)
             redis_client.set("session_dir", self._session_dir)
             redis_client.set("temp_dir", self._temp_dir)
+        else:
+            if (self._config != "{}"):
+                logger.error(
+                    "Internal Config parameters can only be set on the head "
+                    "node. NOTE, this include lru_evict.")
+                self._config == str(dict())
```

Comment:
Why not just raising errors?

Reply:
Yeah, we should just be raising an error here.

Also, lru_evict isn't an internal config parameter from the user's perspective.

Reply:
But `lru_evict` does modify `internal_config` parameters--how should that be handled?

Reply:
I'd just have a separate check for lru_evict flag - if it's set and not head raise an error that says `lru_evict can only be set on head node`

Reply:
I just added it to the massive check block in `worker.py`